### ISSUE TITLE
If my datetime is represented as an integer, all the synthetic values are from 1970

### ISSUE THREAD

**Issue by npatki** [sdv-dev/SDV#1477]
### Environment Details

* SDV version: 1.2.0
* Python version: Any
* Operating System: Any

### Error Description
In some cases, I may have a datetime column represented as an integer such as YYYYMMDD. For example `20230623` encodes today (June 23, 2023).

Even though I am specifying the format correctly in the SDV, it doesn't seem to read in these datetime valuesproperly. In the synthetic data, I usually see years from 1970. It appears to be reading `20230623` as "20230623 seconds after Unix Epoch (Jan 1, 1970)".

### Steps to reproduce

```python
import pandas as pd
from sdv.metadata import SingleTableMetadata
from sdv.single_table import GaussianCopulaSynthesizer

data = pd.DataFrame(data={
    'dt_1': [20230623, 20210405, 20201221, 20190727, 20221107]
})

metadata = SingleTableMetadata.load_from_dict({
    'columns': {
        'dt_1': { 'sdtype': 'datetime', 'datetime_format': '%Y%m%d' },
    }
})

synth = GaussianCopulaSynthesizer(metadata)
synth.fit(data)
synth.sample(num_rows=5)
```

Output: 
<img width="785" alt="image" src="https://github.com/sdv-dev/SDV/assets/3227804/c9b9a751-014c-4436-88c7-e9fed0a07421">

### Workaround
For any datetime columns that are represented as integers, **cast them as objects before training your synthesizer.** The SDV seems to correctly parse object datetimes, as opposed to strings.

From the above example:
```python
# cast any integer datetime columns as objects
data['dt_1'] = data['dt_1'].astype("object")

# now the SDV will correctly read the datetime
synth = GaussianCopulaSynthesizer(metadata)
synth.fit(data)
synth.sample(num_rows=5)
```

<img width="439" alt="image" src="https://github.com/sdv-dev/SDV/assets/3227804/69190677-7664-438b-ab6b-7a4d69b1a287">

**Comment by npatki** [comment]
This will likely be solved by [RDT #657](https://github.com/sdv-dev/RDT/issues/657)

**Comment by npatki** [comment]
This issue has been resolved by the RDT issue and is now working in the SDV 1.2.1 release.
